### ISSUE TITLE
[proposal] Use Poetry as the Python management tool

### ISSUE THREAD

**Issue by haoxins** [alibaba/GraphScope#1160]
I just noticed that we already have `pyproject.toml` inside the Python projects.
Should we consider to use [Poetry](https://github.com/python-poetry/poetry) as the Python build tool?

The benefits will be:
1. More stable dependencies management, such as `lock` file.
2. Built-in virtual env supports
3. Integrate with `lint/format/unittest` tolls, such as `pytest`, `black`, `isort`, and others
4. More easier to add custom build scripts
5. More easier to link local package as dependency

**Comment by siyuan0322** [comment]
Thank you, it's a great tool, we will keep an eye on it. However, it's not the best time to replace the dependency management system now, as our main focus is on new features and user-friendly improvement. It may be introduced in a future release.

Thank you again for your proposal!

**Comment by haoxins** [comment]
> However, it's not the best time to replace the dependency management system now, as our main focus is on new features and user-friendly improvement. It may be introduced in a future release.

Of course, agreed totally. This is not something worth a high priority.
